File: Others/048.Rotate-Image/048.Rotate-Image_v2.py
class Solution:
    def rotate(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: void Do not return anything, modify matrix in-place instead.
        """
        N = len(matrix)
        for i in range(N//2):
            matrix[i],matrix[N-1-i] = matrix[N-1-i],matrix[i]
        for i in range(N):
            for j in range(0,i):
                matrix[i][j],matrix[j][i] = matrix[j][i],matrix[i][j]
                

File: Others/048.Rotate-Image/048.Rotate-Image_v2.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) 
    {
        int n = matrix.size();
        for (int i=0; i<n; i++)
            for (int j=0; j<i; j++)
            {
                swap(matrix[i][j], matrix[j][i]);
            }           
        
        for (int i=0; i<n; i++)
        {
            for (int j=0; j<n/2; j++)
                swap(matrix[i][j], matrix[i][n-1-j]);
        }        
        
    }
};
